### PR TITLE
Add metrics for spilling

### PR DESCRIPTION
**Pull request by bikramSingh91**
Add the following global metrics:
velox.spill_input_bytes
velox.spill_bytes
velox.spill__rows_count
velox.spill_files_count
velox.spill_fill_time_ms
velox.spill_sort_time_ms
velox.spill_serialization_time_ms
velox.spill_disk_writes_count
velox.spill_flush_time_ms
velox.spill_write_time_ms

### CODE REVIEW

File: velox/common/base/Counters.h [hunk 1]
```diff
@@ -66,4 +66,34 @@ constexpr folly::StringPiece kMetricArbitratorArbitrationTimeMs{
 
 constexpr folly::StringPiece kMetricArbitratorFreeCapacityBytes{
     "velox.arbitrator_free_capacity_bytes"};
+
+constexpr folly::StringPiece kMetricSpilledInputBytes{
+    "velox.spill_spilled_input_bytes"};
+
+constexpr folly::StringPiece kMetricSpilledBytes{"velox.spill_spilled_bytes"};
```
Comment (xiaoxmeng) [inline velox/common/base/Counters.h:73]: nit: do we want to call this spill_output_bytes

File: velox/common/base/Counters.h [hunk 2]
```diff
@@ -66,4 +66,34 @@ constexpr folly::StringPiece kMetricArbitratorArbitrationTimeMs{
 
 constexpr folly::StringPiece kMetricArbitratorFreeCapacityBytes{
     "velox.arbitrator_free_capacity_bytes"};
+
+constexpr folly::StringPiece kMetricSpilledInputBytes{
+    "velox.spill_spilled_input_bytes"};
```
Comment (xiaoxmeng) [inline velox/common/base/Counters.h:71]:
```
s/spill_spilled_input_bytes/spill_input_bytes/
```

File: velox/common/base/Counters.h [hunk 3]
```diff
@@ -66,4 +66,34 @@ constexpr folly::StringPiece kMetricArbitratorArbitrationTimeMs{
...
+
+constexpr folly::StringPiece kMetricSpilledBytes{"velox.spill_spilled_bytes"};
+
+constexpr folly::StringPiece kMetricSpilledRowsCount{
+    "velox.spill_spilled_rows_count"};
```
Comment (xiaoxmeng) [inline velox/common/base/Counters.h:76]:
```
s/spill_spilled_rows_count/spill_rows_count
```

File: velox/common/base/Counters.h [hunk 4]
```diff
@@ -66,4 +66,34 @@ constexpr folly::StringPiece kMetricArbitratorArbitrationTimeMs{
...
+
+constexpr folly::StringPiece kMetricSpilledPartitionsCount{
```
Comment (xiaoxmeng) [inline velox/common/base/Counters.h:78]: Not sure if this is useful. We can avoid this and add later if needs.

File: velox/common/base/Counters.h [hunk 5]
```diff
@@ -66,4 +66,34 @@ constexpr folly::StringPiece kMetricArbitratorArbitrationTimeMs{
...
+    "velox.spill_spilled_partitions_count"};
+
+constexpr folly::StringPiece kMetricSpilledFilesCount{
```
Comment (xiaoxmeng) [inline velox/common/base/Counters.h:81]:
```
s/spill_spilled_files_count/spill_files_count/
```

**Review by xiaoxmeng — APPROVED** [review]
@bikramSingh91 LGTM. Thanks!

**Comment by bikramSingh91** [comment]
added documentation

**Pull request closed without merging**